Warning prior to change: 
> /home/michaelgiba/code/github/llama.cpp/tools/mtmd/clip.cpp:3475:1: error: control reaches end of non-void function [-Werror=return-type]
 3475 | }
      | ^

--- a/tools/mtmd/clip.cpp
+++ b/tools/mtmd/clip.cpp
@@ -3468,7 +3468,7 @@ bool clip_image_preprocess(struct clip_ctx * ctx, const clip_image_u8 * img, str
 
         return true;
     } else {
-        GGML_ASSERT(false && "Unknown image preprocessing type");
+        GGML_ABORT("Unknown image preprocessing type");
         return false;
     }
 

--- a/tools/mtmd/clip.cpp
+++ b/tools/mtmd/clip.cpp
@@ -3469,7 +3469,6 @@ bool clip_image_preprocess(struct clip_ctx * ctx, const clip_image_u8 * img, str
         return true;
     } else {
         GGML_ABORT("Unknown image preprocessing type");
-        return false;
     }
 
 }

